lib/dialogs/share/index.jsx

```diff
@@ -116,12 +119,13 @@ export class ShareDialog extends Component {
                   onSubmit={this.onAddCollaborator}
                 >
                   <input
-                    ref={e => (this.collaboratorElement = e)}
-                    type="email"
-                    pattern="[^@]+@[^@]+"
                     className="settings-item-text-input transparent-input"
+                    // Regex to detect valid email
+                    pattern="^[^\s@]+@[^\s@]+\.[^\s@]+$"
```

Comment:
I feel really bad commenting here again, but this pattern does relax some rules but probably rejects valid email address that the previous monster pattern allowed since emails can have more than one `@` sign. I checked and I don't think we have any existing accounts with more than one so it's probably safe to do, but it stood out to me so I asked about it.